pkg/resources/provisioning.cattle.io/v1/cluster/mutator.go

```diff
@@ -133,13 +134,54 @@ func (m *ProvisioningClusterMutator) Admit(request *admission.Request) (*admissi
 		return response, nil
 	}
 
+	if request.Operation == admissionv1.Update {
+		response, err = m.handleDynamicSchemaDrop(request, oldCluster, cluster)
+		if err != nil {
+			return nil, fmt.Errorf("unable to evaluate dynamic schema drop, %w", err)
+		}
+		if response.Result != nil {
+			return response, nil
+		}
+	}
+
 	response.Allowed = true
 	if err = patch.CreatePatch(clusterJSON, cluster, response); err != nil {
 		return nil, fmt.Errorf("failed to create patch: %w", err)
 	}
 	return response, nil
 }
 
+// handleDynamicSchemaDrop watches for provisioning cluster updates, and reinserts the previous value of the
+// dynamicSchemaSpec field for a machine pool if the "provisioning.cattle.io/allow-dynamic-schema-drop" annotation is
+// not present and true on the cluster. If the value of the annotation is true, no mutation is performed.
+func (m *ProvisioningClusterMutator) handleDynamicSchemaDrop(request *admission.Request, oldCluster, cluster *v1.Cluster) (*admissionv1.AdmissionResponse, error) {
```

Comment:
Now that we removed the old and new extraction, I think that this doesn't need to return an error any  more. WDYT?